### PR TITLE
feat(IL): New country Israel

### PR DESCRIPTION
**Pull request by commenthol**
Tries to solve #36.

The test will fail currently as a fix in the parser is needed to solve an issue with Hebrew to Gregorian Calendar conversion.

### CODE REVIEW

File: test/fixtures/IL-2022.json [hunk 1]
```diff
@@ -0,0 +1,290 @@
+[
+  {
+    "date": "2022-01-17 00:00:00 -0600",
+    "start": "2022-01-16T16:00:00.000Z",
```
Comment (oprogramador) [inline test/fixtures/IL-2022.json:4]: On January 16, the sunset in Jerusalem is at 16:59.

File: test/fixtures/IL-2022.json [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    "end": "2022-01-17T16:00:00.000Z",
+    "name": "Tu Bishvat (Arbor Day)",
+    "type": "observance",
+    "rule": "15 Shvat",
+    "_weekday": "Mon"
+  },
+  {
+    "date": "2022-02-14 00:00:00 -0600",
+    "start": "2022-02-13T16:00:00.000Z",
```
Comment (oprogramador) [inline test/fixtures/IL-2022.json:13]: Fast of Esther 2022 is on March 16

https://www.timeanddate.com/holidays/israel/2022

File: test/fixtures/IL-2022.json [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    "date": "2022-01-17 00:00:00 -0600",
+    "start": "2022-01-16T16:00:00.000Z",
+    "end": "2022-01-17T16:00:00.000Z",
+    "name": "Tu Bishvat (Arbor Day)",
+    "type": "observance",
+    "rule": "15 Shvat",
+    "_weekday": "Mon"
+  },
+  {
+    "date": "2022-02-14 00:00:00 -0600",
```
Comment (oprogramador) [inline test/fixtures/IL-2022.json:12]: Fast of Esther 2022 is on March 16

https://www.timeanddate.com/holidays/israel/2022

File: test/fixtures/IL-2022.json [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+    "start": "2022-02-13T16:00:00.000Z",
+    "end": "2022-02-14T16:00:00.000Z",
+    "name": "Fast of Esther",
+    "type": "school",
+    "rule": "13 Adar",
+    "_weekday": "Mon"
+  },
+  {
+    "date": "2022-02-15 00:00:00 -0600",
```
Comment (oprogramador) [inline test/fixtures/IL-2022.json:21]: Purim 2022 is on March 17

https://www.timeanddate.com/holidays/israel/2022

File: test/fixtures/IL-2022.json [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+    "end": "2022-04-16T15:00:00.000Z",
+    "name": "Passover",
+    "type": "public",
+    "rule": "15 Nisan",
+    "_weekday": "Sat"
+  },
+  {
+    "date": "2022-04-17 00:00:00 -0600",
+    "start": "2022-04-16T15:00:00.000Z",
+    "end": "2022-04-21T15:00:00.000Z",
```
Comment (oprogramador) [inline test/fixtures/IL-2022.json:52]: Passover 2022 ends at March 22

https://www.timeanddate.com/holidays/israel/2022

File: test/fixtures/IL-2022.json [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+    "start": "2022-05-04T15:00:00.000Z",
+    "end": "2022-05-05T15:00:00.000Z",
+    "name": "Fallen Soldiers and Victims of Terrorism Remembrance Day",
+    "type": "observance",
+    "note": "National remembrance day, business as usual except places of public entertainment",
+    "rule": "4 Iyyar",
+    "_weekday": "Thu"
+  },
+  {
+    "date": "2022-05-06 00:00:00 -0600",
```
Comment (oprogramador) [inline test/fixtures/IL-2022.json:99]: Independence Day (Yom HaAtzmaut) 2022 is on May 5

File: test/fixtures/IL-2022.json [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+    "date": "2022-06-05 00:00:00 -0600",
+    "start": "2022-06-04T15:00:00.000Z",
+    "end": "2022-06-05T15:00:00.000Z",
+    "name": "Shavuot",
+    "type": "public",
+    "rule": "6 Sivan",
+    "_weekday": "Sun"
+  },
+  {
+    "date": "2022-07-16 00:00:00 -0600",
```
Comment (oprogramador) [inline test/fixtures/IL-2022.json:154]: 17th of Tammuz 2022 is on July 17

https://www.timeanddate.com/holidays/israel/2022

File: test/fixtures/IL-2022.json [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+    "date": "2022-07-28 00:00:00 -0600",
+    "start": "2022-07-27T15:00:00.000Z",
+    "end": "2022-07-28T15:00:00.000Z",
+    "name": "Jabotinsky Day",
+    "type": "public",
+    "rule": "29 Tamuz",
+    "_weekday": "Thu"
+  },
+  {
+    "date": "2022-08-06 00:00:00 -0600",
```
Comment (oprogramador) [inline test/fixtures/IL-2022.json:172]: Tisha B'Av 2020 is on July 7

https://www.timeanddate.com/holidays/israel/2022

**Comment by oprogramador** [comment]
We need to fix tests before merging.

File: test/fixtures/IL-2015.json [hunk 9]
```diff
@@ -0,0 +1,308 @@
+[
```
Comment (oprogramador) [inline test/fixtures/IL-2015.json:1]: How are the dates calculated?

Fixtures are OK for testing but we need to have a reliable way to calculate dates for any year in the future or in the past.

**Comment by commenthol** [comment]
Hi @oprogramador, Thanks for your review. I use a "custom" calendar calculated from hebcal. I suspect that the conversion which is done there needs some fixing. Currently I don't have the time to look into this topic, but I won't forget about it.

**Comment by commenthol** [comment]
fix needs parser update for passing the tests

**Comment by oprogramador** [comment]
> fix needs parser update for passing the tests see [commenthol/date-holidays-parser#37](https://github.com/commenthol/date-holidays-parser/pull/37)

Thanks for https://github.com/commenthol/date-holidays-parser/pull/37. I left some comments there and I also see a build failing.

**Comment by commenthol** [comment]
Hi @oprogramador,
The parser is now up-to-date with the required fixes.
Tests are passing as well with the different node versions.
Maybe you'd like to review the changes made on the holiday rules.
Thanks

**Comment by oprogramador** [comment]
> Hi @oprogramador, The parser is now up-to-date with the required fixes. Tests are passing as well with the different node versions. Maybe you'd like to review the changes made on the holiday rules. Thanks

Thanks. I left some new comments on https://github.com/commenthol/date-holidays-parser/pull/37 after it had been merged and I open two my PRs: https://github.com/commenthol/date-holidays-parser/pull/38 & https://github.com/commenthol/date-holidays-parser/pull/39.

File: data/countries/IL.yaml [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+    langs:
+      - he
+    zones:
+      - Asia/Jerusalem
+    dayoff: saturday
+    days:
+      1 Tishrei:
+        name:
+          he: ראש השנה
+          en: Rosh Hashanah
```
Comment (oprogramador) [inline data/countries/IL.yaml:16]:
```suggestion
          en: Rosh Hashanah (Yom Teruah)
```

File: data/countries/IL.yaml [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+      2 Tishrei:
+        name:
+          he: ראש השנה יום 2
+          en: Rosh Hashanah (Day 2)
+      3 Tishrei if saturday then next sunday:
+        name:
+          he: צום גדליה
+          en: Fast of Gedaliah
```
Comment (oprogramador) [inline data/countries/IL.yaml:24]:
```suggestion
          en: Fast of Gedaliah (Tzom Gedalya)
```

File: data/countries/IL.yaml (hunk 10 above)
Comment (commenthol) [reply data/countries/IL.yaml:16]: ok

File: data/countries/IL.yaml (hunk 11 above)
Comment (commenthol) [reply data/countries/IL.yaml:24]: ok

File: test/fixtures/IL-2015.json (hunk 9 above)
Comment (commenthol) [reply test/fixtures/IL-2015.json:1]: Please check CONTRIBUTING.md

File: test/fixtures/IL-2022.json (hunk 1 above)
Comment (commenthol) [reply test/fixtures/IL-2022.json:4]: Please check section "Dates in Hebrew calendar" of specification.md

File: test/fixtures/IL-2022.json (hunk 3 above)
Comment (commenthol) [reply test/fixtures/IL-2022.json]: Should be solved

File: test/fixtures/IL-2022.json (hunk 4 above)
Comment (commenthol) [reply test/fixtures/IL-2022.json]: Should be solved with update

File: test/fixtures/IL-2022.json (hunk 5 above)
Comment (commenthol) [reply test/fixtures/IL-2022.json:52]: Should be solved

File: test/fixtures/IL-2022.json (hunk 6 above)
Comment (commenthol) [reply test/fixtures/IL-2022.json]: Should be solved

File: test/fixtures/IL-2022.json (hunk 7 above)
Comment (commenthol) [reply test/fixtures/IL-2022.json]: Should be solved

File: test/fixtures/IL-2022.json (hunk 8 above)
Comment (commenthol) [reply test/fixtures/IL-2022.json]: Should be solved

File: data/countries/IL.yaml [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+        name:
+          en: Passover (intermediate days)
+          he: חול המועד פסח
+        type: school
+        note: School holiday, collective paid leave in many businesses and government offices
+      21 Nisan:
+        name:
+          en: Seventh day of Passover
+          he: שביעי של פסח
+      22 Nisan:
```
Comment (oprogramador) [inline data/countries/IL.yaml:108]: it should be 21 Nisan

Comment (oprogramador) [reply data/countries/IL.yaml:108]: I found:
- Simchat Torah: 22nd (outside of Israel 23rd) day of Tishrei
- Mimouna: begins 21st day of Nisan in Israel; 22nd day of Nisan outside of Israel

so logically either both should be according to the date in Israel or both according to the date outside of Israel.

Comment (oprogramador) [reply data/countries/IL.yaml:108]: However, the remaining holidays in this file look good to me.

Comment (oprogramador) [reply data/countries/IL.yaml:108]: Beside Wikipedia, I found:
- https://www.theshulofnewyork.org/event/passover-mimouna-2019.html
- https://www.jesusboat.com/all-about-mimouna-post-passover/
- https://www.sukkatshalom.org/event/end-of-passover-moroccan-mimouna.html

**Pull request merged**